File: 1_two_sum/TwoSum.java
class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash =  new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int num = target - nums[i];

            if (hash.containsKey(num) && hash.get(num) != i) {
                return new int[]{i, hash.get(num)};
            }

            hash.put(nums[i], i);
        }

        return new int[]{};
    }
}

File: 1_two_sum/TwoSum.cpp
class TwoSum {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> numToIndex;
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (numToIndex.find(complement) != numToIndex.end()) {
                return {numToIndex[complement], i};
            }
            numToIndex[nums[i]] = i;
        }
        throw std::invalid_argument("No solution found");
    }
};